prov/sm2/src/sm2_progress.c

```diff
@@ -192,8 +329,8 @@ static int sm2_start_common(struct sm2_ep *ep,
 			rx_entry->iov, rx_entry->count, &total_len, ep);
 		break;
 	case sm2_proto_cma:
-		err = sm2_progress_cma(xfer_entry, rx_entry->iov,
-				       rx_entry->count, &total_len, ep, 0);
+		err = sm2_progress_cma(ep, xfer_entry, rx_entry, &total_len, 0,
+				       &ipc_host_to_dev);
```

Comment:
I would like to add one if statement here that handles all the device to host stuff for this function, and returns early... so we do not have to mess up the rest of the common path with "!if device to host"

Reply:
Done